Feature/file manager

StylishPanther opened the pull request:
## Background

This commit is a skeleton code including class and its method definition and declaration.


## Test Result

```
Running main() from C:\Users\User\source\repos\SSDTestShell\packages\gmock.1.11.0\lib\native\src\gtest\src\gtest_main.cc
[==========] Running 7 tests from 2 test suites.
[----------] Global test environment set-up.
[----------] 6 tests from FileManagerTest
[ RUN      ] FileManagerTest.file_manager_test_open
[       OK ] FileManagerTest.file_manager_test_open (0 ms)
[ RUN      ] FileManagerTest.file_manager_test_init
[       OK ] FileManagerTest.file_manager_test_init (0 ms)
[ RUN      ] FileManagerTest.file_manager_test_read
[       OK ] FileManagerTest.file_manager_test_read (0 ms)
[ RUN      ] FileManagerTest.file_manager_test_write
[       OK ] FileManagerTest.file_manager_test_write (0 ms)
[ RUN      ] FileManagerTest.file_manager_test_write_during_read
[       OK ] FileManagerTest.file_manager_test_write_during_read (0 ms)
[ RUN      ] FileManagerTest.file_manager_test_close
[       OK ] FileManagerTest.file_manager_test_close (0 ms)
[----------] 6 tests from FileManagerTest (1 ms total)
```

StylishPanther committed:
[feat] FileManager skeleton code is implemented

StylishPanther committed:
[feat] Update FileManager class skeleton code and its unit test

miryo9yd commented on SSDManager/FileManager.h:
```diff
@@ -0,0 +1,16 @@
+{
+public:
+	FileManager();
+	std::fstream* open(std::string);
```
변수 명이 누락된 것 같습니다. 검토 부탁 드립니다. : )

StylishPanther replied:
넵! 매개변수 이름까지 명시하는게 좋을 것 같습니다! 리뷰 감사합니다.

miryo9yd reviewed (changes requested):
FlashManager.h open 함수에 인자 이름이 없는 부분이 있어 검토 요청 드려요 ~ : )

Kanaish commented on SSDManagerTest/FileManagerTest.cpp:
```diff
@@ -1,10 +1,42 @@
+
+TEST(FileManagerTest, file_manager_test_read) {
+    FileManager fm;
+    std::fstream * fs = fm.open("test.txt");
```
using namespace std를 써서 가독성을 높이는 것은 어떠신가요?

StylishPanther replied:
말씀하신대로 test 파일에서는 써도 괜찮을 것 같습니다!
다음 커밋에 반영하겠습니다.

jiyeon-08 commented on SSDManager/FileManager.h:
```diff
@@ -0,0 +1,16 @@
+	unsigned int read(std::fstream* fs, int index);
+	bool write(std::fstream* fs, int index, unsigned int value);
+	bool write(std::fstream* fs, unsigned int value);
+	bool close(std::fstream* fs);
```
수고하셨습니다~ 코딩 컨벤션 룰에 맞게 tab -> 2space로 변경해주시면 좋을 것 같습니다

StylishPanther replied:
넵! 리뷰 감사합니다.

StylishPanther committed:
[refacto] Correction for wrong notation

1) Adapting std namespace in unit test file
2) Add parameter name on init method in Filemanager
3) Lint correction for filemanager header file

StylishPanther committed:
[refactor] Correction for wrong notation

1) Adapting std namespace in unit test file
2) Add parameter name on init method in Filemanager
3) Lint correction for filemanager header file

bsjn16sw commented on SSDManager/FileManager.h:
```diff
@@ -0,0 +1,16 @@
+	std::fstream* init(std::string name);
+	unsigned int read(std::fstream* fs, int index);
+	bool write(std::fstream* fs, int index, unsigned int value);
+	bool write(std::fstream* fs, unsigned int value);
```
`index` 가 없는 `write` 는 언제 사용되나요??

StylishPanther replied:
index가 없는 write는 read 하는 과정에서 result.txt에 read한 값을 쓸 때 사용됩니다.
일단 편의상 overloading을 통해서 구현하고, 추후 Refactoring을 통해 명시성을 강화하려고 합니다.

bsjn16sw reviewed (approved):
고생 많으셨습니다! 👍👍

miryo9yd reviewed (approved):
고생하셨습니다. 코드리뷰 완료했어요 .~

Kanaish commented on SSDManager/FileManager.h:
```diff
@@ -3,14 +3,13 @@
-	bool write(std::fstream* fs, unsigned int value);
-	bool close(std::fstream* fs);
+    FileManager();
+    std::fstream* init(std::string name);
```
cpp에서 std 빠진 것 확인했습니다. 감사합니다~
근데 아직 header에는 남아있습니다 확인 부탁드려요 :)

StylishPanther replied:
오프라인에서 토의한 결과,
google test project와 sdd project는 서로 독립적인 프로그램이기 때문에, 가독성을 위해서
google test project에서는 using namespace 를 사용하고, sdd project에서는 namespace를 사용하지 않는 것으로 결론 되었습니다.

wonmyth reviewed (approved):
빠른 구현 감사합니다.
리뷰 완료했습니다.

Kanaish reviewed (commented):
오 빠른 개발 및 수정 감사합니다.
prefix를 [refactor] 바꾸면 좋을 것 같습니다 :)

StylishPanther committed:
[refactor] Correction for wrong notation

1) Correct commit message to fit in commit message style.

Kanaish reviewed (approved):
commit 메시지 수정까지 확인 완료했습니다
감사합니다 :)

StylishPanther commented:
모든 리뷰어들이 승인하여 Merge 합니다.
